### DISCUSSION TITLE
Can't get the example to work.

### DISCUSSION THREAD

**Discussion by mournblade** [focus-shift/jollyday#528] [Q&A] [answered]
Hi, trying to migrate from the "old" jollyday, but having problems with the example code. Using 0.27 via maven, but when I run this method:
`private HolidayManager getHolidayManager() {
        if(holidayManager == null) {
            holidayManager = HolidayManager.getInstance(ManagerParameters.create(SWEDEN));
        }
        return holidayManager;
    }`

I get an error:

> java.lang.IllegalStateException: Cannot instantiate datasource instance because there is no implementations
>	at de.focus_shift.jollyday.core.datasource.ConfigurationServiceManager.instantiateDataSource(ConfigurationServiceManager.java:33)

Any idea what this is? Pointers appreciated.

**Comment by derTobsch** [comment, marked as answer]
You need to use the Jackson or jaxb dependency and not the core.

**Reply by mournblade** [reply]
Allright, I see it now, thanks. Total noob question, but why? I'm not using it to parse anything, literally just want the holidays as in the example? Is it to read config files from disc?

**Reply by derTobsch** [reply]
The holidays are described in XML files. So we need Jackson or jaxb to parse them.

**Comment by JcontrerasV1997** [comment]
Missing configuration 'manager.impl'. Cannot create manager.

public static boolean isHoliday(LocalDate date, String countryCode) {
       final HolidayManager holidayManager = HolidayManager.getInstance(ManagerParameters.create(HolidayCalendar.GERMANY));
       final Set<Holiday> holidays = holidayManager.getHolidays(Year.of(2022));
       return true;
    }
    
    Please help me, what could be happening?

**Reply by derTobsch** [reply]
you need to add a xml dependency like jollyday-jaxb or jollyday-jackson, see https://github.com/focus-shift/jollyday?tab=readme-ov-file#xml-binding-libraries

**Comment by YSavitski** [comment]
Hi @derTobsch 

I've already made a post, but probably not in the proper issue.

Can you please have a look at https://github.com/focus-shift/jollyday/issues/398#issuecomment-3571549916 ?

Many thanks in advance

**Reply by derTobsch** [reply]
Hey @YSavitski,
can you provide a minimum example? You need to add at least the jaxb or the jackson dependency that transitively brings with it the core dependency. But it is recommended to add both to your pom e.g. directly.

**Reply by YSavitski** [reply]
> Hey @YSavitski, can you provide a minimum example? You need to add at least the jaxb or the jackson dependency that transitively brings with it the core dependency. But it is recommended to add both to your pom e.g. directly.

Hey @derTobsch 

I provided example in another issues, please have a look: https://github.com/focus-shift/jollyday/issues/398#issuecomment-3571549916

but also got response that it make sense to add explicitly `jollyday-core` into my project

Many thanks
